### PR TITLE
[buildozer] Add ability to print separators in output

### PR DESCRIPTION
**Pull request by nataliejameson**
Sometimes we want to print out multiple fields in print commands. If any of those fields are a list, it causes the number of columns to be variable, and it's kind of hard to parse things out. This also applies to when arrays are flattened. If there is a space in any of the elements, we can't tell one element from multiple elements. This patch allows us to specify how things should be printed out. 

Also added a simple integration test that passed with bazel test edit/... . Lemme know if there's a different place to write those integration style tests, as I didn't see anything.

### CODE REVIEW

**Comment by philipjameson** [comment]
I should be in the fb CLA org now.

**Comment by philipjameson** [comment]
@googlebot One more once

**Comment by laurentlb** [comment]
Thanks!

(sorry, I'll be on vacation for a few days)

**Comment by philipjameson** [comment]
I'd lean towards json if that's what we'd want to do as it's a little more universally parsable. I was actually toying with just taking the normal protobuf objects and marshaling them with the json marshaler instead, but it seems like maybe the protobuf object has some internal details and that like that get a bit messy (e.g. I think there was a 'bazel_quoted_string' object or something like that at one point).

**Comment by philipjameson** [comment]
Holy cow I let this sit for a long time. @laurentlb ?

**Comment by laurentlb** [comment]
@vladmos, what do you think?

I think it would be more valuable to have a JSON output. The separators flags look like a partial workaround and might not be enough (separators might be ambiguous unless we also have proper escaping of the content).

Possibly a command `buildozer print_json` or a flag to select json output

**Comment by philipjameson** [comment]
Yeah, I think something like --json should work. The initial concern was that for large changes, you can't really read iteratively (I'm using this in a tool that outputs a fair chunk of data), but that may be negligible. Should this just be the protobuf format, or something different?

**Comment by philipjameson** [comment]
Added in a --output_format field (can rename if desired), added a few more tests, and removed the extra flags I had before. This now just prints out the json proto format.

**Comment by hick209** [comment]
Any updates on this? I see this here got merged https://github.com/bazelbuild/buildtools/pull/612, should this PR be closed? Else what's the hold on this PR?

**Comment by oreflow** [comment]
+1 on https://github.com/bazelbuild/buildtools/pull/313#issuecomment-3715328622 Looks like this was resolved to a large extent already, so closing it. feel free to reopen if it's still relevant
